iFixit/react-commerce
# Add collection empty states

This PR adds an empty state when the collection is empty or when the search product no results. Moreover we're adding also an initial loading state, tough this would not be visible when the collection is rendered server side (right now we're still rendering the collection client side).



## QA

1. Visit the Vercel PR preview
2. Visit an empty collection page (e.g `/parts`)
3. Verify that the collection empty state is rendered
    ![Screen Shot 2021-08-04 at 09 55 25](https://user-images.githubusercontent.com/4640135/128143678-ddca8fbc-48bf-47e0-b533-9651cc1a275d.png)
4. Visit a collection that has products (e.g. `/collections/macbook`) and filter so that there is no result (e.g. type random characters on the search bar like "dfsd")
5. Verify that the "No results found" empty state is rendered
    ![Screen Shot 2021-08-04 at 09 58 19](https://user-images.githubusercontent.com/4640135/128144106-a0785ea2-1008-493e-9543-0bdb35b4391c.png) 

Closes #49
